src/algebra/group_power/lemmas.lean

```diff
@@ -49,17 +49,13 @@ lemma inv_of_pow (m : M) [invertible m] (n : ℕ) [invertible (m ^ n)] :
 lemma is_unit.pow {m : M} (n : ℕ) : is_unit m → is_unit (m ^ n) :=
 λ ⟨u, hu⟩, ⟨u ^ n, by simp *⟩
 
-lemma is_unit_pos_pow_iff {M : Type*} [comm_monoid M] {m : M} {n : ℕ} (h : 0 < n) :
+lemma is_unit_pos_pow_iff {m : M} {n : ℕ} (h : 0 < n) :
   is_unit (m ^ n) ↔ is_unit m :=
 begin
-  obtain ⟨p, rfl⟩ := nat.exists_eq_succ_of_ne_zero h.ne',
-  refine ⟨λ h, _, is_unit.pow _⟩,
-  obtain ⟨⟨k, k', hk, hk'⟩, h⟩ := h,
-  rw [units.coe_mk] at h,
-  refine ⟨⟨m, m ^ p * k', _, _⟩, _⟩,
-  { rw [←mul_assoc, ←pow_succ, ←h, hk] },
-  { rw [mul_right_comm, ←pow_succ', ←h, hk] },
-  { exact units.coe_mk _ _ _ _ }
+  refine ⟨_, λ h, h.pow _⟩,
+  obtain ⟨p, rfl : n = p + 1⟩ := nat.exists_eq_succ_of_ne_zero h.ne',
```

Comment:
Do you think it's worth adding
```
@[simp] lemma is_unit_pow_succ_iff {m : M} {n : ℕ} : is_unit (m ^ n.succ) ↔ is_unit m :=
```
from the other PR, since that's an intermediate step of this proof anyway?